core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -514,20 +510,19 @@ public Double getFeatureVariableDouble(@Nonnull String featureKey,
             return null;
         }
 
-        String variableValue = getFeatureVariableValueForType(
-            featureKey,
-            variableKey,
-            userId,
-            attributes,
-            FeatureVariable.VariableType.DOUBLE
-        );
-        if (variableValue != null) {
-            try {
-                return Double.parseDouble(variableValue);
-            } catch (NumberFormatException exception) {
-                logger.error("NumberFormatException while trying to parse \"" + variableValue +
-                    "\" as Double. " + exception);
-            }
+        Double variableValue = null;
+        try {
+            variableValue = getFeatureVariableValueForType(
+                featureKey,
+                variableKey,
+                userId,
+                attributes,
+                FeatureVariable.VariableType.DOUBLE
+            );
+            return variableValue;
+        } catch (Exception exception) {
+            logger.error("NumberFormatException while trying to parse \"" + variableValue +
+                "\" as Double. " + exception);
```

Comment:
@mnoman09 this and a [other](https://github.com/optimizely/java-sdk/pull/270/files#diff-9719ba1e8c37710e7b92b94f0330e552R577) [log](https://github.com/optimizely/java-sdk/pull/270/files#diff-9719ba1e8c37710e7b92b94f0330e552R729) statements in this PR append the exception as a string on log message. the logger should be passed the exception object as a last argument